### PR TITLE
Update Snakemake language config

### PR DESCRIPTION
**Pull request by TeemuSailynoja**
1. Updated the `file-types` argument in Snakemake language definition to properly detect the language for `Snakefile`.
2. Added a formatter command for Snakemake using `snakefmt`.

### CODE REVIEW

File: languages.toml [hunk 1]
```diff
@@ -3840,11 +3840,14 @@ source = { git = "https://github.com/Decurity/tree-sitter-circom", rev = "021505
...
 scope = "source.snakemake"
 roots = ["Snakefile", "config.yaml", "environment.yaml", "workflow/"]
-file-types = ["smk", "Snakefile"]
+file-types = ["smk", { glob = "Snakefile" } ]
 comment-tokens = ["#", "##"]
 indent = { tab-width = 2, unit = "  " }
 language-servers = ["pylsp" ]
 
+[language.formatter]
+command = "snakefmt"
```
Comment (TeemuSailynoja) [inline languages.toml:3849]: Snakefmt is the formatter provided by the Snakemake Github organisation: https://github.com/snakemake/snakefmt

File: languages.toml [hunk 2]
```diff
@@ -3840,11 +3840,14 @@ source = { git = "https://github.com/Decurity/tree-sitter-circom", rev = "021505
...
+args = ["-"]
```
Comment (TeemuSailynoja) [inline languages.toml:3850]: This was not so clearly documented, but in the warning callout at the start of the Github [Readme](https://github.com/snakemake/snakefmt?tab=readme-ov-file#snakefmt), they state:

> You can also pipe the file in from stdin, which will print it to the screen.

Later, in the last example of [Basic Usage](https://github.com/snakemake/snakefmt?tab=readme-ov-file#basic-usage) they show how to write the output to stdout with
```Snakemake
snakefmt - < Snakefile
```
It turns out, that by removing the `< Snakefile` part above, `snakefmt` takes input from stdin and sends the output to stdout.

File: languages.toml [hunk 3]
```diff
@@ -3840,11 +3840,14 @@ source = { git = "https://github.com/Decurity/tree-sitter-circom", rev = "021505
 name = "snakemake"
 scope = "source.snakemake"
 roots = ["Snakefile", "config.yaml", "environment.yaml", "workflow/"]
-file-types = ["smk", "Snakefile"]
+file-types = ["smk", { glob = "Snakefile" } ]
```
Comment (TeemuSailynoja) [inline languages.toml:3843]: In the original commit, "Sankefile" was handled as a suffix, when instead Helix should look at the full filename to apply the language.  Like `Make` and `Makefile`, the `Snakemake` pipeline is defined in `Snakefile`. I added the required `glob` part.

**Review by the-mikedavis — APPROVED** [review]

**Review by the-mikedavis — APPROVED** [review]

**Comment by TeemuSailynoja** [comment]
@pascalkuthe, sorry to bother you, but I saw that you reviewed the original Snakemake PR. Could you check this one too, or assign someone for the review?

Thank you!

**Pull request merged**